src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -530,5 +535,95 @@ public static string[] GetLogicalDrives()
         {
             return DriveInfoInternal.GetLogicalDrives();
         }
+
+        internal static string? GetLinkTarget(ReadOnlySpan<char> linkPath, bool isDirectory) => Interop.Sys.ReadLink(linkPath);
+
+        internal static void CreateSymbolicLink(string path, string pathToTarget, bool isDirectory)
+        {
+            string pathToTargetFullPath = PathInternal.GetLinkTargetFullPath(path, pathToTarget);
+
+            // Fail if the target exists but is not consistent with the expected filesystem entry type
+            if (Interop.Sys.Stat(pathToTargetFullPath, out Interop.Sys.FileStatus targetInfo) == 0)
+            {
+                if (isDirectory != ((targetInfo.Mode & Interop.Sys.FileTypes.S_IFMT) == Interop.Sys.FileTypes.S_IFDIR))
+                {
+                    throw new IOException(SR.Format(SR.IO_InconsistentLinkType, path));
+                }
+            }
+
+            Interop.CheckIo(Interop.Sys.SymLink(pathToTarget, path), path, isDirectory);
+        }
+
+        internal static FileSystemInfo? ResolveLinkTarget(string linkPath, bool returnFinalTarget, bool isDirectory)
+        {
+            // throws if the current link file does not exist
+            Interop.CheckIo(Interop.Sys.LStat(linkPath, out Interop.Sys.FileStatus info), linkPath, isDirectory);
+
+            if ((info.Mode & Interop.Sys.FileTypes.S_IFMT) != Interop.Sys.FileTypes.S_IFLNK)
+            {
+                return null;
+            }
+
+            ValueStringBuilder sb = new(stackalloc char[Interop.DefaultPathBufferSize]);
```

Comment:
`ValueStringBuilder` instance is being disposed only for the lucky path. When an exception is being thrown, it's not.